packages/protocol/lib/web3-utils.ts

```diff
@@ -387,6 +393,28 @@ export function getFunctionSelectorsForContract(contract: any, contractName: str
   return selectors
 }
 
+export function checkInheritance(baseContractName: string, derivativeContractArtifact: any) {
+  return derivativeContractArtifact.ast.nodes.find(
+    (astNode: any) => astNode.nodeType === 'ImportDirective' && (astNode.file as string).endsWith(`${baseContractName}.sol`)
+  ) !== undefined
+}
+
+export function checkInheritanceDeep(baseContractName: string, derivativeContractArtifact: any, artifacts: any) {
+  const isImport = (astNode: any) => astNode.nodeType === 'ImportDirective'
+  const imports: any[] = derivativeContractArtifact.ast.nodes.filter((astNode: any) => isImport(astNode))
+  while (imports.length) { // BFS 
+    const importedContractName = (imports.pop().file as string).split('/').pop().split('.')[0]
+    if (importedContractName ===  baseContractName) {
+      return true
+    }
+    const importedContractArtifact = artifacts instanceof BuildArtifacts ? 
```

Comment:
Why do we need to support two different types of artifacts objects? Is there a way to simplify this?

Reply:
This is because we use both truffle artifacts and openzeppelin artifacts in our tooling

Reply:
since we need to perform this check in places where both artifacts are used, I'm not sure if there's a way to consolidate without doing some more major surgery to only use one type of artifacts object throughout our tooling.